### DISCUSSION TITLE
Pricing: Do you charge for audios totally in silence or audios with only noise?

### DISCUSSION THREAD

**Discussion by davasquez** [deepgram/community#293] [General help] [answered]
Will I get charge if I try to transcribe audios totally in silence and/or audios with only noise?

**Comment by jjmaldonis** [comment]
Hey @davasquez, Deepgram doesn't know the audio doesn't contain silence or noise until after the transcription process is invoked, so you will get charged for transcribing silent audio and noise. There are open source libraries for both silence detection and noise suppression if you are interested and can pre-process your audio before sending it to Deepgram.

Are you expecting significant amounts of silence or only-noise in your audio files?

**Reply by davasquez** [reply]
hi @jjmaldonis , thanks for your reply, yes, we are expecting quite a lot of audio files with silence or only-noise. The problem is that we must use streaming. We tried without streaming but it was not possible to use in our application given that it is a real-time application.

**Reply by jjmaldonis** [reply, marked as answer]
Got it - since this is streaming you may be in luck. If you know in advance when users will stop talking (for example, if the user is on hold on a phone call), then you can keep the websocket alive by using a [Keep Alive message](https://developers.deepgram.com/reference/streaming#stream-keepalive).

This is how the pricing works:
- the user (you) stops sending audio data and doesn't send keepalive: Deepgram will close the websocket after 12 sec, with no charges for the time without audio data
- the user stops sending audio and sends keepalives: Deepgram won't close the websocket, with no charges for the time without audio
- the user continues to send audio data and sends keepalives: Deepgram won't close and will continue to transcribe the audio, and the user will be charge for the audio transcribed
- the user sends a keepalive followed by silent audio: the user will be charged for the silent audio
